server/aesdsocket.c

```diff
@@ -20,150 +21,334 @@
 #include <unistd.h>
 #include <arpa/inet.h>
 #include <signal.h>
+#include <sys/queue.h>
+#include <pthread.h>
+#include <sys/time.h>
 
-// Macro directives
+// macro definition 
 #define TOTAL_MSG	3	
 #define ERROR		1
 #define SUCCESS		0
 #define MYPORT		9000
 #define DEF_FILEPATH	"/var/tmp/aesdsocketdata"
+#define BUFFER_LEN	200
 
-int sockfd = 0;
-char *sendbuffer = NULL;
-int acceptedfd = 0;
-int fd = 0;
-sigset_t mask;
+typedef struct
+{
+    pthread_t thread;
+    int fd;
+    int acceptedfd;
+	struct in_addr sin_addr;
+    bool thread_complete_success;
+    bool complete_status_flag;
 
-//filename and path in one variable
-const char *filepath = DEF_FILEPATH;
+}thread_data;
 
-//Function:	static void signal_handler(int signo)
-//Inputs:	signo - Signal number
 
-static void closeall(void)
+typedef struct slist_data_s slist_data_t;
+struct slist_data_s{
+
+    thread_data params;
+    SLIST_ENTRY(slist_data_s) entries;
+};
+
+int sockfd;
+int exit_on_signal=0;
+
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
@MuktaDarekar think of a more descriptive name for this.  What is it locking?

Reply:
updated.